<table align="center" width=100%>
    <tr>
        <td width="15%">
            <img src="homework.png">
        </td>
        <td>
            <div align="center">
                <font color="#21618C" size=8px>
                    <b> Take-Home <br>(Day 1)
                    </b>
                </font>
            </div>
        </td>
    </tr>
</table>

**Import the required libraries**

In [1]:
# type your code here
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import scipy.stats as stats

### Let's begin with some hands-on practice exercises

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>1. The rainfall (in mm) in the city is recorded for 10 days. Find the rainfall value under which 60% of the rainfall would lie.</b>
                </font>
            </div>
        </td>
    </tr>
</table>

Given data:

        rainfall (in mm) = [86, 74, 90, 108, 65, 84, 75, 92, 102, 93]

In [2]:
rainfall=[86, 74, 90, 108, 65, 84, 75, 92, 102, 93]

In [3]:
np.quantile(rainfall,q=0.6)

90.80000000000001

##### INFERENCE:-
    The rainfall value under which 60% of the rainfall would lie is 90.8

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>2. The performance of John and Jack in the monthly exams is given below. Identify the most consistent student. </b>
                </font>
            </div>
        </td>
    </tr>
</table>

Given data:

        John = [42, 38, 47, 35, 39, 46]
        Jack = [32, 36, 46, 49, 28, 30]

In [4]:
# type your code here
John=[42, 38, 47, 35, 39, 46]
Jack=[32, 36, 46, 49, 28, 30]

In [5]:
np.mean(John)

41.166666666666664

In [6]:
np.std(John)

4.29793231940921

In [7]:
np.mean(Jack)

36.833333333333336

In [8]:
np.std(Jack)

7.966945601811413

##### INFERENCE:-
            The most consistent student is John because his Standard Deviation is less compared to Jack

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>3. The amount of Calcium, Potassium and Iron in the chocolate cookies of 5 different brands is collected. Find out which mineral can be neglected while comparing the different cookie brands?</b>
                </font>
            </div>
        </td>
    </tr>
</table>

Given data:

        Calcium (in mg) = [132, 138.56, 147.2, 145.6, 139]
        Potassium (in mg) = [122.2, 116, 106.6, 119.52, 128]
        Iron (in mg) = [2.96, 3.02, 3.01, 2.99, 2.93]

In [9]:
# type your code here
Calcium=[132, 138.56, 147.2, 145.6, 139]
Potassium=[122.2, 116, 106.6, 119.52, 128]
Iron=[2.96, 3.02, 3.01, 2.99, 2.93]

In [10]:
np.std(Calcium)/np.mean(Calcium)

0.03888221107112609

In [11]:
np.std(Potassium)/np.mean(Potassium)

0.060026949580646076

In [12]:
np.std(Iron)/np.mean(Iron)

0.011101908354960974

##### INFERENCE:-
    The mineral that can be neglected while comparing the different cookie brands is Iron beacuse it has Least Co-efficient of Variance

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>4. On an online payment gateway, on average 9 out of 25 customers are getting an error while redeeming the gift coupon. A random sample of 40 customers is selected. Find the probability that at most 16 customers will get the same error. 
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [13]:
# type your code here
p=9/25
n=40
r=16
stats.binom.cdf(r,n,p)

0.7577762236447019

##### INFERENCE:-
    The probability that atmost 16 customers will get the same error is 0.11

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>5. The survey by road safety department states that the in the Oneonta city the average speed of vehicle is 65 mph and the standard deviation is 9 mph. Find the probability that a randomly selected vehicle has a speed greater than 78 mph.
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [14]:
# type your code here
mu=65
sigma=9
x=78
z=(x-mu)/sigma
stats.norm.sf(z)

0.07430699822760586

##### INFERENCE:-
    The probability that a randomly selected vehicle has a speed greater than 78 mph is 0.074

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>6. A wrist-watch company wants to perform a quality check before launching their next-generation k22-digital watch. The production team has delivered 30 watches with number tags as 1,2,...30 on each watch for the quality check. Pick 5 watches randomly to check the quality. </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [15]:
data=np.arange(1,31,1)
data

array([ 1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16, 17,
       18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30])

In [16]:
np.random.choice(data,size=5,replace=True)

array([30,  1, 24, 16, 22])

##### INFERENCE:-
     5 watches are picked randomly with Sampling with Replacement to check the quality

In [17]:
np.random.choice(data,size=5,replace=False)

array([17,  7, 20, 25, 29])

##### INFERENCE:-
    5 watches are picked randomly with Sampling without Replacement to check the quality

<table align="left">
    <tr>
        <td width="6%">
            <img src="question_icon.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>7.The fill amount in 2-liter soft drink bottles is normally distributed, with a mean of 2.0 liters and a standard deviation of 0.05 liter. If the bottles contain less than 95% of the listed net content (1.90 liters, in our case), the manufacturer may be subject to penalty by the state office of consumer affairs. Bottles that have a net content above 2.1 liters may cause excess spillage upon opening. What is the proportion of bottles that will contain
                      
a) between 1.9 and 2.0 liters
                        
b) between 1.9 and 2.1 liters
                        
c) below 1.9 liters or above 2.1 liters
                        
d) At least how much soft drink(in litres) is contained in 99% of the bottles? </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [18]:
# type your code here
#Given:-
mu=2
sigma=0.05

In [19]:
#a) between 1.9 and 2.0 liters
x1=1.9
x2=2
z1=(x1-mu)/sigma
z2=(x2-mu)/sigma
print(z1)
print(z2)
stats.norm.cdf(z2)-stats.norm.cdf(z1)

-2.0000000000000018
0.0


0.4772498680518209

##### INFERENCE:-
    The proportion of bottles that will contain between 1.9 and 2.0 liters is 0.47

In [20]:
#b) between 1.9 and 2.1 liters
x1=1.9
x2=2.1
z1=(x1-mu)/sigma
z2=(x2-mu)/sigma
print(z1)
print(z2)
stats.norm.cdf(z2)-stats.norm.cdf(z1)

-2.0000000000000018
2.0000000000000018


0.9544997361036418

##### INFERENCE:-
    The proportion of bottles that will contain between 1.9 and 2.1 liters is 0.95

In [21]:
#c) below 1.9 liters or above 2.1 liters
x1=1.9
x2=2.1
z1=(x1-mu)/sigma
z2=(x2-mu)/sigma
print(z1)
print(z2)
1-(stats.norm.cdf(z2)-stats.norm.cdf(z1))

-2.0000000000000018
2.0000000000000018


0.045500263896358195

##### INFERENCE:-
The proportion of bottles that will contain below 1.9 liters or above 2.1 liters is 0.99

In [22]:
#d) At least how much soft drink(in litres) is contained in 99% of the bottles?

In [23]:
x=1.99
z=(x-mu)/sigma
stats.norm.sf(z)

0.5792597094391031

##### INFERENCE:-
    Atleast how much soft drink(in litres) is contained in 99% of the bottles is 0.99